### PR TITLE
Add Support for Encoding in PBES2

### PR DESCRIPTION
**Pull request by Tookmund**
With the release of OpenSSL v3, openssl now defaults to "AES password based encryption (PBES2 with PBKDF2 and AES-256-CBC) for private keys and certificates, the PBKDF2 and MAC key derivation iteration count of PKCS12_DEFAULT_ITER (currently 2048), and MAC algorithm HMAC with SHA2-256."
https://www.openssl.org/docs/man3.0/man3/PKCS12_create.html
All other formats are now considered legacy.

This PR builds on the configuration work done in #39 , simply adding support for PBES2 as an algorithm choice.
This doesn't support encoding in every algorithm that could be used for PBES2, but only the exact format used by OpenSSL.
Other formats are left as a future enhancement, and will likely require a rethink of the interface used to select algorithms in #39.

Additionally this solves #44 by bumping the `x/crypto` dependency to `0.9.0`. This appears to work without any changes required.

### CODE REVIEW

**Review by maraino — CHANGES_REQUESTED** [review]
I haven't looked at the full PR, only at the KDF parameters. I am not a go-pkcs12 mainteiner.

File: safebags.go [hunk 1]
```diff
@@ -77,6 +126,35 @@ func encodePkcs8ShroudedKeyBag(rand io.Reader, privateKey interface{}, password
 	return asn1Data, nil
 }
 
+func encodePBES2Params(salt []byte, rand io.Reader) (paramBytes []byte, err error) {
+	iv := make([]byte, 16)
+	if _, err = rand.Read(iv); err != nil {
```
[the remark this answers is not in the document]
Comment (Tookmund) [reply safebags.go]: Thanks! I've found a number of places where we read from rand without checking and corrected them all to this construction.

**Comment by Tookmund** [comment]
I've defaulted to 2048 in most places, since that's still what OpenSSL uses, but I've increased SafeEncode to the standard 600,000 iterations.

**Comment by Tookmund** [comment]
Closed in favor of #48

**Pull request closed without merging**